code/controllers/subsystem/polling.dm

```diff
@@ -121,83 +129,89 @@ SUBSYSTEM_DEF(polling)
+	if(!(amount_to_pick > 0))
+		return new_poll.signed_up
+	for(var/pick in 1 to amount_to_pick)
+		new_poll.chosen_candidates += pick_n_take(new_poll.signed_up)
+	if(announce_chosen)
+		new_poll.announce_chosen(group)
+	if(new_poll.chosen_candidates.len == 1)
+		var/chosen_one = pick(new_poll.chosen_candidates)
+		return chosen_one
+	return new_poll.chosen_candidates
+
+/datum/controller/subsystem/polling/proc/poll_ghost_candidates(question, role, check_jobban, poll_time = 30 SECONDS, ignore_category = null, flashwindow = TRUE, alert_pic, jump_target, role_name_text, list/custom_response_messages,\
+	start_signed_up = FALSE, amount_to_pick = 0, chat_text_border_icon, announce_chosen = TRUE)
 	var/list/candidates = list()
 	if(!(GLOB.ghost_role_flags & GHOSTROLE_STATION_SENTIENCE))
-		return candidates
-
+		return
 	for(var/mob/dead/observer/ghost_player in GLOB.player_list)
 		candidates += ghost_player
-
-	return poll_candidates(question, role, check_jobban, poll_time, ignore_category, flashwindow, candidates, pic_source, role_name_text)
-
-/datum/controller/subsystem/polling/proc/poll_ghost_candidates_for_mob(question, role, check_jobban, poll_time = 30 SECONDS, mob/target_mob, ignore_category = null, flashwindow = TRUE, pic_source, role_name_text)
-	var/static/list/mob/currently_polling_mobs = list()
-
-	if(currently_polling_mobs.Find(target_mob))
-		return list()
-
-	currently_polling_mobs += target_mob
-
-	var/list/possible_candidates = poll_ghost_candidates(question, role, check_jobban, poll_time, ignore_category, flashwindow, pic_source, role_name_text)
-
-	currently_polling_mobs -= target_mob
-	if(!target_mob || QDELETED(target_mob) || !target_mob.loc)
-		return list()
-
-	return possible_candidates
-
-/datum/controller/subsystem/polling/proc/poll_ghost_candidates_for_mobs(question, role, check_jobban, poll_time = 30 SECONDS, list/mobs, ignore_category = null, flashwindow = TRUE, pic_source, role_name_text)
-	var/list/candidate_list = poll_ghost_candidates(question, role, check_jobban, poll_time, ignore_category, flashwindow, pic_source, role_name_text)
-
-	for(var/mob/potential_mob as anything in mobs)
-		if(QDELETED(potential_mob) || !potential_mob.loc)
-			mobs -= potential_mob
-
-	if(!length(mobs))
+	return poll_candidates(question, role, check_jobban, poll_time, ignore_category, flashwindow, candidates, alert_pic, jump_target, role_name_text, custom_response_messages, start_signed_up, amount_to_pick, chat_text_border_icon, announce_chosen)
+
+/datum/controller/subsystem/polling/proc/poll_ghosts_for_target(question, role, check_jobban, poll_time = 30 SECONDS, atom/movable/checked_target, ignore_category = null, flashwindow = TRUE, alert_pic, jump_target, role_name_text, list/custom_response_messages,\
+	start_signed_up = FALSE, chat_text_border_icon, announce_chosen = TRUE)
```

Comment:
the formatting on the procs in this file looks kinda dumb, if you want to multiline these you should go all the way
```suggestion
/datum/controller/subsystem/polling/proc/poll_ghosts_for_target(
	question,
	role,
	check_jobban,
	poll_time = 30 SECONDS,
	atom/movable/checked_target,
	ignore_category = null,
	flashwindow = TRUE,
	alert_pic,
	jump_target,
	role_name_text,
	list/custom_response_messages,
	start_signed_up = FALSE,
	chat_text_border_icon, announce_chosen = TRUE,
)
```